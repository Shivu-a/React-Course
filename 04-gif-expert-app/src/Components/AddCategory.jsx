import React, { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onChangedInput = (e) => {
    setInputValue(e.target.value);
  };

  const onSubmit = (e) => {
    // Prevenimos el default
    e.preventDefault();

    // Creamos una variable con el input
    const newInputValue = inputValue.trim();

    // Chequeamos el input
    if (newInputValue.length <= 1) return;

    // Creamos la nueva categoria y reseteamos el value
    onNewCategory(newInputValue);
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Ingrese su busqueda"
        value={inputValue}
        onChange={onChangedInput}
      />
    </form>
  );
};
