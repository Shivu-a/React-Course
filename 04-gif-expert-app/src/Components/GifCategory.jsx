import React from "react";
import PropTypes from "prop-types";

export const GifCategory = ({ url, id, title }) => {
  return (
    <div className="card">
      <p>{title}</p>
      <img src={url} alt={title} />
    </div>
  );
};

GifCategory.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};
