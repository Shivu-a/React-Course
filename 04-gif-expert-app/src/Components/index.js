export * from "./AddCategory";
export * from "./GifCategory";
export * from "./GifGrid";
