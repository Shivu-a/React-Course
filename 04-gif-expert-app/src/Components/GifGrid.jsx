import React from "react";

import { GifCategory } from "./GifCategory";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <div className="gifCategory">
      <h3>{category}</h3>
      {isLoading && <div className="loading"></div>}
      <div className="card-grid">
        {images.map((image) => (
          <GifCategory key={image.id} {...image} />
        ))}
      </div>
    </div>
  );
};
