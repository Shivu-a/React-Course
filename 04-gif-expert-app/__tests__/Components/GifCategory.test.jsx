import { render, screen } from "@testing-library/react";

import { GifCategory } from "../../src/Components/GifCategory";

describe("Tests en GifCategory", () => {
  const image = {
    key: "Test",
    url: "https://test.com/",
    title: "Test",
  };

  test("Deberia de hacer match con el snapshot", () => {
    const { container } = render(<GifCategory key={image.key} {...image} />);

    expect(container).toMatchSnapshot();
  });

  test("Deberia de mostrar con el URL y el alt indicado", () => {
    render(<GifCategory key={image.key} {...image} />);

    const { src, alt } = screen.getByRole("img");

    expect(src).toBe(image.url);
    expect(alt).toBe(image.title);
  });

  test("Titulo deberia existir en el componente", () => {
    render(<GifCategory key={image.key} {...image} />);

    expect(screen.getByText).toBeTruthy;
  });
});
