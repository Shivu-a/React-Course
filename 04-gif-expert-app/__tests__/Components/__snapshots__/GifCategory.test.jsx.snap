// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests en GifCategory Deberia de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      Test
    </p>
    <img
      alt="Test"
      src="https://test.com/"
    />
  </div>
</div>
`;
